geopandas/base.py

```diff
@@ -5003,6 +5003,66 @@ def sample_points(self, size, method="uniform", seed=None, rng=None, **kwargs):
 
         return GeoSeries(result, name="sampled_points", crs=self.crs, index=self.index)
 
+    def build_area(self, node=True):
+        """Creates an areal geometry formed by the constituent linework.
+
+        Builds areas from the GeoSeries that contain linework which represents the edges
+        of a planar graph. Any geometry type may be provided as input; only the
+        constituent lines and rings will be used to create the output polygons.
+
+        This function converts inner rings into holes. To turn inner rings into polygons
+        as well, use polygonize.
+
+        Unless you know that the input GeoSeries represents a planar graph with a clean
+        topology (e.g. there is a node on both lines where they intersect), it is
+        recommended to use ``node=True`` which performs noding prior to building areal
+        geometry. Using ``node=False`` will provide performance benefits but may result
+        in incorrect polygons if the input is not of the proper topology.
+
+        If the input linework crosses, this function will produce invalid polygons. Use
+        :meth:`GeoSeries.make_valid` to ensure valid geometries.
```

Comment:
Is this also true with the default of `node=True` in our case? Because I would expect the noding will fix those crossing lines?

Reply:
I suppose the individual lines in the input can still cross, and the unary union for `node=True` doesn't necessarily fix that? (I don't know how unary union typically handles invalid input)

Reply:
I will change it to "may" but in reality, I wasn't able to produce invalid polygons no matter how crossing lines I created. If you pass in an invalid polygon, it raises (e.g. `GEOSException: TopologyException: side location conflict at 0.5 0.5. This can occur if the input geometry is invalid.`).